### PR TITLE
axes3d panning

### PR DESCRIPTION
**Pull request by emilyfy**
## PR Summary
not sure why axes3d panning with middle scroll button is not implemented. I'm putting it here and it works as far as I tested. 

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and `pydocstyle<4` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by QuLogic** [comment]
Can you write a What's new entry for this? You also have flake8 issues that would need to be fixed.

**Comment by emilyfy** [comment]
@QuLogic sorry about that, I'm not very familiar with the proper formatting. I fixed the mentioned flake8 issues, but what's a `what's new entry`?

**Comment by WeatherGod** [comment]
For new features, we add a "what's new" entry that gets collated into a
single document for release. The instructions are here:
https://matplotlib.org/3.3.1/users/next_whats_new/README.html

**Comment by tacaswell** [comment]
Thanks for this @emilyfy !  I suspect the reason it was not implemented is no one had implemented it yet ;).  

I checked this out and verified that it works as expected.

Given that this whole block of code is untested I am not going to block merging on that, but if you are feeling motivated, following the pattern in https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/testing/widgets.py is the path to go.

---

I agree that this is a big enough feature added that we need to make sure it gets advertised in the whats new!

**Review by tacaswell — APPROVED** [review]

**Comment by emilyfy** [comment]
I added a short what's new entry.
Sorry again, but I'm not sure what exactly to do for testing the block of code. Is there perhaps a more related example I can refer to, where they test the rotation or zooming in the same _on_move function maybe?

**Review by QuLogic — APPROVED** [review]
This works pretty well for me. I'm not sure we really have any good way to test 3D interactivity.

File: lib/matplotlib/testing/widgets.py [hunk 1]
```diff
@@ -18,31 +18,35 @@ def get_ax():
     return ax
 
 
-def do_event(tool, etype, button=1, xdata=0, ydata=0, key=None, step=1):
-    """
-    Trigger an event
+def mock_event(ax, button=1, xdata=0, ydata=0, key=None, step=1):
```
Comment (timhoffm) [inline lib/matplotlib/testing/widgets.py:21]: Git diff looks more complicated than it is. This just extracts the event creation out of `do_event()`.

**Comment by timhoffm** [comment]
@emilyfy Sorry, for being a bit confusing. Writing the test was actually a bit more difficult than thought, because the `do_event()` mechanism cannot be applied to mouse operations targeting a 3D Axes. I took the liberty of pushing a test to your branch.

ping @tacaswell @QuLogic for re-review/merge.

**Review by QuLogic — APPROVED** [review]

**Comment by timhoffm** [comment]
Thanks @emilyfy! Congratulations on your first contribution to Matplotlib. We'd be happy to see you back some time.

**Pull request merged**